crates/polars-core/src/chunked_array/ops/split.rs

```diff
@@ -0,0 +1,47 @@
+pub struct SplitNChars<'a> {
+    s: &'a str,
+    n: usize,
+    keep_remainder: bool,
+}
+
+impl<'a> Iterator for SplitNChars<'a> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let single_char_limit = if self.keep_remainder { 2 } else { 1 };
+        if self.n >= single_char_limit {
```

Comment:
what if f keep_remainder is true but we're on the last character?

Reply:
polars code, thoroughly tested before us